### PR TITLE
Add New UI Tests App for critical tests

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :scroll: Description

This PR adds crash recovery and corrupted envelope recovery UI Tests for Sentry Android. 

## :bulb: Motivation and Context

- Part of https://github.com/getsentry/projects/issues/165

## :green_heart: How did you test it?
locally, ci

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

## :crystal_ball: Todo

- [x] Add workflow
- [x] Add make command for local execution

#skip-changelog

### CODE REVIEW

File: sentry-android-integration-tests/sentry-uitest-android-critical/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    signingConfigs {
+        getByName("debug") {
+            // Debug config remains unchanged
+        }
+    }
+
+    defaultConfig {
+        applicationId = "io.sentry.uitest.android.critical"
+        minSdk = 21
```
Comment (stefanosiano) [inline sentry-android-integration-tests/sentry-uitest-android-critical/build.gradle.kts:20]:
```suggestion
        minSdk = Config.Android.minSdkVersionCompose
```

**Review by stefanosiano — APPROVED** [review]
good!

**Comment by krystofwoldrich** [comment]
I'm currently waiting for the job to run multiple times (50-100x) to ensure it's stable, before merging.

**Comment by krystofwoldrich** [comment]
Run from stability testing https://github.com/getsentry/sentry-java/actions/runs/11439418099/job/31828821106?pr=3802 (27 runs, 1 emulator boot timeout)

**Pull request merged**